test/core/symbol/SymbolCrypto.spec.ts

```diff
@@ -0,0 +1,115 @@
+        // Act:
+        const encoded = encode(sender.privateKey, recipient.publicKey, messageBytes, customerIV);
+        const decoded = decode(recipient.privateKey, sender.publicKey, encoded);
+
+        // Assert:
+        expect(decoded).deep.equal(messageBytes);
+
+        return encoded;
+    };
+
+    describe('Round Trip Test', () => {
+        it('Can encode and decode message using random IV', () => {
+            encodeAndDecodeAssert();
+        });
+
+        it('Can encode and decode message using provided IV', () => {
+            encodeAndDecodeAssert(customerIV);
+        });
+    });
+
+    describe('Edge Case Tests', () => {
+        it('Can encode a message but failed decode with wrong key', () => {
+            // Arrange:
+            const invalidKey = Key.createFromHex('57F7DA205008026C776CB6AED843393F04CD458E0AA2D9F1D5F31A402072B2D6');
+
+            // Act:
+            const encoded = encodeAndDecodeAssert();
+
+            // Assert:
+            expect(() => {
+                decode(recipient.privateKey, invalidKey, encoded);
+            }).to.throw('Unsupported state or unable to authenticate data');
+            expect(() => {
+                decode(invalidKey, sender.publicKey, encoded);
+            }).to.throw('Unsupported state or unable to authenticate data');
+            expect(() => {
+                decode(invalidKey, invalidKey, encoded);
+            }).to.throw('Unsupported state or unable to authenticate data');
+        });
+
+        it('Encode throws if using empty key', () => {
+            // Act+Assert:
+            expect(() => {
+                encode(emptyKey, recipient.publicKey, messageBytes);
+            }).to.throw();
+            expect(() => {
+                encode(sender.privateKey, emptyKey, messageBytes);
+            }).to.throw();
+        });
+
+        it('Decode throws if using empty key', () => {
+            // Arrange:
+            const encoded = encodeAndDecodeAssert();
+            // Act+Assert:
+            expect(() => {
+                decode(recipient.privateKey, emptyKey, encoded);
+            }).to.throw();
+            expect(() => {
+                encode(sender.privateKey, emptyKey, messageBytes);
+            }).to.throw();
+        });
+
+        it('Encode throws if using empty message', () => {
+            // Act+Assert:
+            expect(() => {
+                encode(sender.privateKey, recipient.publicKey, new Uint8Array());
+            }).to.throw();
+        });
+
+        it('Encode throws if using invalid IV', () => {
+            // Act+Assert:
+            expect(() => {
+                encode(sender.privateKey, recipient.publicKey, messageBytes, new Uint8Array(1));
+            }).to.throw();
+        });
+
+        it('decode throws if using invalid payload', () => {
+            // Arrange:
+            const encoded = encodeAndDecodeAssert();
+
+            // Act+Assert:
+            expect(encoded.length).greaterThan(28);
```

Comment:
not sure purpose of this ?

Reply:
To make sure the payload contains a 12 bytes IV and the 16 bytes tag?